Performance

sfshaza2 opened the pull request:
Staged:
https://sz-flutter.firebaseapp.com/docs/get-started/test-drive

https://sz-flutter.firebaseapp.com/docs/perf

Filip and Yuqian, 

You'll see that this is just the beginning of the changes. If you would like me to publish the app size stuff sooner, rather than later, I can temporarily remove the TBD performance pages that haven't been written yet.

liyuqian commented on src/docs/perf/app-size.md:
````diff
@@ -0,0 +1,87 @@
+```shell
+flutter build apk --target-platform=android-arm64
+```
+
````
Same here. 64bit is a little bigger

> Here's an example output
> ```
> Built build/app/outputs/apk/release/app-release.apk (4.6MB).
> ```

liyuqian commented on src/docs/perf/app-size.md:
````diff
@@ -0,0 +1,87 @@
+```shell
+flutter build ios && tar -zcf build/app.ipa build/ios/iphoneos/Runner.app && ls -lh build/app.ipa
+```
+
````
Same here:
```
-rw-r--r--  1 liyuqian  primarygroup   8.3M Oct 25 13:47 build/app.ipa
```

eseidelGoogle commented:
FYI @devoncarew who has been thinking about perf tooling, including application size.

sfshaza2 commented:
@liyuqian and @devoncarew, I've made some updates:

- As I'd like to publish this sooner, rather than later, I've removed the unwritten TBD pages.
- I've incorporated feedback from @liyuqian.

Still staged at: 
https://sz-flutter.firebaseapp.com/docs/get-started/test-drive

https://sz-flutter.firebaseapp.com/docs/perf

devoncarew commented on src/docs/perf/rendering/index.md:
```diff
@@ -0,0 +1,28 @@
+Flutter applications are performant by default,
+so you only need to avoid common pitfalls to achieve
+excellent performance instead of needing to micro-optimize
+with complicated profiling tools.
```
I would strike the part about complicated profiling tools, and instead focus on Flutter being fast by default.

devoncarew commented on src/docs/perf/rendering/index.md:
```diff
@@ -0,0 +1,28 @@
+
+If you _are_ seeing janky (non smooth) animations, make
+**sure** that you are profiling performance with an
+app built in _profile_ mode.
```
Do we describe somewhere (here or in other docs) how to run in profile mode?

sfshaza2 replied:
Yes, the link to "Flutter's build modes" has more info.

jacob314 commented on src/docs/development/tools/devtools/inspector.md:
```diff
@@ -135,6 +135,6 @@ of the Flutter inspector.
 [Debugging Flutter apps]: /docs/testing/debugging
-[The performance overlay]:/docs/testing/ui-performance#the-performance-overlay
-[Flutter performance profiling]: /docs/testing/ui-performance
+[The performance overlay]:/docs/perf/rendering/ui-performance#the-performance-overlay
```
nit: missing space before `/`

jacob314 commented on src/docs/perf/power.md:
```diff
@@ -0,0 +1,5 @@
+---
+title: Preserving your battery
+---
+
```
@terrylucas @kenzieschmoll @DaveShuckerow 
Currently the best we can do in DevTools would be to visualize what fraction of the time the app was keeping the dart isolates active.
IOS and Android both provide some battery usage charts in their debugging tools.

jacob314 commented on src/docs/perf/rendering/index.md:
```diff
@@ -0,0 +1,28 @@
+Thanks, in part, to Flutter's Skia engine and its ability
+to quickly create and dispose of widgets, 
+Flutter applications are performant by default,
+so you only need to avoid common pitfalls to achieve
```
common pitfalls:
* rebuilding far more of the app than expected each frame (link to widget rebuild tracker feature in IntelliJ).
* Building a large list of children directly instead of using a ListView
* See https://github.com/InMatrix/perf_diagnosis_demo/tree/master/lib
  for more examples

liyuqian commented on src/docs/perf/app-size.md:
````diff
@@ -60,6 +73,12 @@ run the following:
 flutter build ios && tar -zcf build/app.ipa build/ios/iphoneos/Runner.app && ls -lh build/app.ipa
 ```
 
+The resulting IPA file for this example is 8.3 MB.
````
Shall we be more specific?

The resulting IPA file for our `example/hello_world` app at the moment is 8.3 MB.

liyuqian commented on src/_includes/run-profile.md:
```diff
@@ -1,5 +1,7 @@
 ## Profile or release runs
 
+
+
```
Are these new lines intended?

liyuqian commented on src/docs/perf/app-size.md:
````diff
@@ -0,0 +1,107 @@
+
+```shell
+flutter build apk --split-per-abi
+```
````
Oops, I forgot this one:

```
Built build/app/outputs/apk/release/app-armeabi-v7a-release.apk (4.2MB).
Built build/app/outputs/apk/release/app-arm64-v8a-release.apk (4.6MB).
```

sfshaza2 commented on src/docs/perf/rendering/index.md:
```diff
@@ -0,0 +1,28 @@
+
+Rendering animations in your app is one of the most cited
+topics of interest when it comes to measuring performance. 
+Thanks, in part, to Flutter's Skia engine and its ability
```
It's correct either way, but I took the edit.  ;)
